lib/floe/workflow/state.rb

```diff
@@ -4,32 +4,31 @@ module Floe
   class Workflow
     class State
       include Logging
+      include ValidationMixin
 
       class << self
-        def build!(workflow, name, payload)
+        def build!(workflow, full_name, payload)
```

Comment:
@kbrock do we care about name vs full_name?  Could we keep it as name but have it be the full path instead?

Reply:
There are a few spots that reference the state's name and want the shorter name.
I've introduced short_name, so I guess we still needed this concept.

So we now have short_name, name (full name), and long_name (type: name)
Maybe we want to drop long_name at a later time